crates/polars-core/src/chunked_array/upstream_traits.rs

```diff
@@ -234,9 +234,23 @@ impl FromIterator<Option<Series>> for ListChunked {
                                 builder.append_null();
                             }
                             builder.append_series(first_s).unwrap();
-
                             for opt_s in it {
-                                builder.append_opt_series(opt_s.as_ref()).unwrap();
+                                match opt_s {
+                                    Some(ref s) => {
+                                        // Empty AnyValue lists (or lists of nulls) have dtype Null.
+                                        // To append them, we cast them to the first value's inner dtype.
```

Comment:
I don't want an extra branch in the `FromIterator` case. Here we should assume that the types are correct (we should add a `debug_assert` on that). I think we should solve this in the code path that produces different inner types.